config/runtime/server.go

```diff
@@ -319,13 +320,19 @@ func newBackend(evalCtx *hcl.EvalContext, backendCtx hcl.Body, log *logrus.Entry
 		beConf.Name = name
 	}
 
+	scheme, hostname, origin := splitBackendOrigin(evalCtx, backendCtx)
+
 	tc := &transport.Config{
 		BackendName:            beConf.Name,
 		DisableCertValidation:  beConf.DisableCertValidation,
 		DisableConnectionReuse: beConf.DisableConnectionReuse,
 		HTTP2:                  beConf.HTTP2,
 		NoProxyFromEnv:         ignoreProxyEnv,
 		MaxConnections:         beConf.MaxConnections,
+
+		Hostname: hostname,
```

Comment:
To configure the health check I prefer an own configuration struct. The health check should be configured independent from transport config.
The transport gets configured per roundtrip on backend level and setting the origin once here to pass this attribute to the health probes could be a bit confusing.

Reply:
Removed passing of `origin` etc. in this location and instead fully configured a `Probe` struct and passed this struct to a new field `ProbeOpts` in the `transport.Config` struct. Like this the `NewProbe()` function can be completely reduced to a single line calling a goroutine with `go b.transportConf.ProbeOpts.probe()`. 

The `Probe` struct has also been adjusted accordingly. Instead of having singular entries for `failure_threshold`, `period` and `timeout` it now simply receives a `health_check.ParsedOptions` struct which contains the three of them. Furthermore, while the old `Probe` struct received a `Config` struct which ended up only being used for its `Scheme` and `Origin` entries to create a new request once at the beginning of `probe()`, the new struct receives a new request with the same origin directly, eliminating the need for the `Config` struct entirely.